### DISCUSSION TITLE
Is it possible to show the documentation of the currently selected candidate with lsp?

### DISCUSSION THREAD

**Discussion by azimut** [company-mode/company-mode#1384] [Q&A]
- More precisely to show the documentation on the "echo area" (eldoc?)
- This works for me with elisp on *scratch* for example. But not on a mode using lsp. (tried a couple of modes with their lsp backends: go, erlang, c)
- Is this an expected behavior? a bug? something that could be configured from the company side?

**Comment by dgutov** [comment]
AFAICS, `lsp-mode` has no support for `:company-docsig`. So we don't have the info to use in the `meta` frontend.

This can have its pluses and minutes: for pluses, eglot can continue showing the info it showed before (outside of current completion).

Perhaps @yyoncho has a better answer.

**Reply by azimut** [reply]
I see, I read a bit of this now https://github.com/emacs-lsp/lsp-mode/issues/3201

Since there is no :company-docsig, there is a way to have :company-doc-buffer (?) into the echo area?

---

Also, just to illustrate my use case. I am using Go and what `company-box` returns is just a small blurb. But given that the signature of each suggestion tends to be really wide it makes the popup documentation appear way to right side. And I feel it would be less annoying look a little bit down instead. 

![image](https://user-images.githubusercontent.com/2867036/234288495-de101375-b8ca-4d02-85a7-687e1a50f2e1.png)

**Reply by dgutov** [reply]
It would be a good idea to file a feature request for `lsp-mode` with this exact description. They will probably know better how to format the doc string to one-liner.

Something else: `company-posframe` has a feature where the contents of the aforementioned "docsig" are shown in the popup, under all completions. This might work a little better for your usage too (or not, of course).
